### DISCUSSION TITLE
Need help in calibration process

### DISCUSSION THREAD

**Discussion by nbpilot007** [jyjblrd/Mocap-Drones#54] [Q&A]
Hi @jyjblrd 
It's a fantastic project and what's more amazing is that you made it open source. I'm trying to do something similar but I'm stuck during the camera calibration process.
Can you please share the order in which the calibration needs to be run? There are a couple of buttons on the GUI but I'm not able to figure out which process to follow first and so on.
I have done the camera pose calibration and I think it turned out fine. But I'm not sure what to do next.

Edit: I read this [comment](https://github.com/jyjblrd/Low-Cost-Mocap/discussions/48#discussioncomment-9628558) and tried the calibration. After running Acquire floor, my cameras rotate around the axis and then upon clicking on Set origin, I get an error. Attaching the same. 

![Screenshot from 2024-06-23 16-20-26](https://github.com/jyjblrd/Low-Cost-Mocap/assets/173601409/2b976fa6-d4e0-4492-916e-6fcf0c3b46dc)

Please help. Thanks in advance.

**Comment by jyjblrd** [comment]
Should be:
1. Capture many points using a wand with two markers set a predetermined distance apart. Then calibrate scale
2. Capture many points of the floor. Then calibrate the floor
3. Capture a single point of the origin. Then calibrate origin

Please let me know if that works

**Reply by nbpilot007** [reply]
It still didn't work. I'm writing the exact steps that I followed. Please correct me if I did it wrong

1. Clicked on "Start (Collect points ....)", moved the wand with one marker, captured around 200 points, then clicked on "Calculate camera pose with 201 points".
After doing this, next step is to Set scale with two markers. But the set scale button is disabled until I hit the "Live Triangulation Start" button.
2. I started "Live Triangulation", then moved the wand with two markers at set distance, then clicked on "Set Scale" and stopped Live Triangulation.
Now the Set Scale, Acquire Floor and Set origin buttons are enabled even while Live triangulation is not running.
3. Now I click on "Start (Collect points ....)" and then move the wand with single marker on the entire floor, capture some 100 points, then hit stop and then click on "Acquire Floor".
After doing this, one of the cameras is at the origin location (the same issue that I was facing earlier and the image I attached).
4. Now I click on "Start (Collect points ....)" and then keep the wand with single marker on the origin point on the floor, capture around 10 points, then hit stop and then click on "Set Origin".
Now all my cameras go below the floor.

I followed the same process but capturing different number of points each time but the result is the same. Please let me know if I'm making any silly errors while doing the calibration.

Thanks in advance

**Reply by nbpilot007** [reply]
This is what I got: 

![Screenshot from 2024-06-29 16-51-06](https://github.com/jyjblrd/Low-Cost-Mocap/assets/173601409/009c1de6-49a3-468f-87d9-4248b56b2807)

**Reply by OmkarShr** [reply]
Hey I am facing the exact same issue you described, getting a decent camera pose calibration and set scale does shrink the stuff decently but the aquire floor puts one camra at the origin and the result does not even make sense.Im also getting error for set origin saying ObjectPoint is empty ..Please guide and help

**Comment by nbpilot007** [comment]
@jyjblrd 
Did you face similar issues when you were caliberating it for you self? if yes then what did you do? any help with whats happening here
Thanks
